_projects/nested-languages.md

```diff
@@ -24,20 +24,51 @@ tags:
   - Core
   - Bears
 ---
-Multiple programming languages can coexist in a single source file. coala would
+Generally, a single source file contains code for exactly one programming language.
+But multiple programming languages can also coexist in a single source file. coala would
 benefit from the ability to determine the programming language of each part of
-a file. If we have e.g. a PHP file, it is likely that we'd have chunks of html
+a file. If we have example a PHP file, it is likely that we'd have chunks of html
 inside a few of them. We would like coala to be able to still run meaningful
 analysis on those files.
 Other examples include:
 
-* awk inside bash
-* JSON inside Javascript
-* HTML, LaTeX or others inside Jinja2
+* JSON inside Javascript is valid in some cases, for example JSONFormatBear for coala.
+* Web templating - languages such as PHP or JSP files mix code into HTML.
+* Cython, it is a superset of the Python language that additionally supports
+  calling C functions and declaring C types on variables and class attributes.
+* We can call Python code from Java using Jython, which is a Python implementation in Java.
+* reStructuredText and Markdown language specify different language for codeblock.
 
-This project is about enabling coala to deal with those situations and allow
-people to write code analysis similar to how they already do it while being
-applicable to the right locations at the right files.
+  For example
+
+```python
+s = "Python syntax highlighting"
+print s
+```
+In the above example we have a code block which is embedded into markdown and language used to write a code block is python.
```

Comment:
Line too long, tests fail here.

Reply:
ya thats by mistake but it is fine now

Reply:
It is not fine now.  80 cpl is our preferred length.